packages/components/select-v2/src/useSelect.ts

```diff
@@ -204,10 +204,10 @@ const useSelect = (props: SelectV2Props, emit: SelectV2EmitFn) => {
     const regexp = new RegExp(escapeStringRegexp(query), 'i')
     const isFilterMethodValid =
       props.filterable && isFunction(props.filterMethod)
-    const isRemoteMethodValid =
+    const isRemoteMethodValid = () =>
       props.filterable && props.remote && isFunction(props.remoteMethod)
     const isValidOption = (o: Option): boolean => {
-      if (isFilterMethodValid || isRemoteMethodValid) return true
+      if (isFilterMethodValid || isRemoteMethodValid()) return true
```

Comment:
There is indeed extra trigger because of `remoteMethod`:
Calling `onTrigger` in the `watchEffect` debug option will print the key `remoteMethod`, (maybe vue can't check the old-new changes of this type (?))

There is also the same issue with `filterMethod`, we can make the same change as `remoteMethod`.
Consider using `toRaw` to understand what's going on:
```suggestion
props.filterable && isFunction(toRaw(props).filterMethod)
    const isRemoteMethodValid =
props.filterable && props.remote && isFunction(toRaw(props).remoteMethod)
const isValidOption = (o: Option): boolean => {
      if (isFilterMethodValid || isRemoteMethodValid) return true
```

Reply:
Change to toRaw
![image](https://github.com/user-attachments/assets/bdd97a42-9833-4025-871e-5cb521593c3f)